pkg/sql/pg_catalog.go

```diff
@@ -1695,32 +1628,19 @@ func getOIDFromConstraint(
 	return oid
 }
 
-var pgCatalogSharedDescriptionTable = virtualSchemaTable{
-	comment: `shared object comments
-https://www.postgresql.org/docs/9.5/catalog-pg-shdescription.html`,
-	schema: vtable.PGCatalogSharedDescription,
-	populate: func(ctx context.Context, p *planner, _ catalog.DatabaseDescriptor, addRow func(...tree.Datum) error) error {
-		// See comment above - could make this more efficient if necessary.
-		comments, err := getComments(ctx, p)
-		if err != nil {
-			return err
-		}
-		for _, comment := range comments {
-			commentType := catalogkeys.CommentType(tree.MustBeDInt(comment[3]))
-			if commentType != catalogkeys.DatabaseCommentType {
-				// Only database comments are exported in this table.
-				continue
-			}
-			classOid := tree.NewDOid(catconstants.PgCatalogDatabaseTableID)
-			objID := descpb.ID(tree.MustBeDInt(comment[0]))
-			if err := addRow(
-				tableOid(objID),
-				classOid,
-				comment[2]); err != nil {
-				return err
-			}
-		}
-		return nil
+// Database comments.
+// https://www.postgresql.org/docs/9.5/catalog-pg-shdescription.html,
+var pgCatalogSharedDescriptionView = virtualSchemaView{
+	// TODO(knz): add an explanatory comment here when this issue is fixed:
+	// https://github.com/cockroachdb/cockroach/issues/95427
+	schema: `
+CREATE VIEW pg_catalog.pg_shdescription AS SELECT objoid, classoid, description
+FROM "".crdb_internal.kv_catalog_comments
+WHERE classoid = ` + strconv.Itoa(catconstants.PgCatalogDatabaseTableID) + `:::oid`,
```

Comment:
if you want to make this even more efficient you could add a virtual partial index 
```
`INDEX(classoid) WHERE classoid = `  + strconv.Itoa(catconstants.PgCatalogDatabaseTableID) + `:::oid`
```

and in that one call `GetAllDatabases` instead of `GetAll`. It'll do a namespace join, so an extra round-trip, but it might read a heck of a lot less data and produce a heck of a lot fewer rows to filter in some cases.